src/modules/session-recorder/SessionRecorder.ts

```diff
@@ -130,12 +136,31 @@ export class SessionRecorder {
     this.#interceptNetworkRequests((event) => {
       this.#addEvent(RQSessionEventType.NETWORK, event);
     });
+
+    if (this.#options.localStorage || this.#options.sessionStorage) {
+      if (this.#options.localStorage) {
+        Storage.addListener(this.#handleStorageEvent, StorageType.LOCAL);
+      }
+      if (this.#options.sessionStorage) {
+        Storage.addListener(this.#handleStorageEvent, StorageType.SESSION);
+      }
+      Storage.initStorageCapture(true, this.#options.localStorage, this.#options.sessionStorage);
+    }
```

Comment:
If `addListener` and `initStorageCapture` are combined then `if (this.#options.localStorage || this.#options.sessionStorage)` can be removed.

```suggestion

      if (this.#options.localStorage) {
        Storage.addListener(this.#handleStorageEvent, StorageType.LOCAL);
      }
      if (this.#options.sessionStorage) {
        Storage.addListener(this.#handleStorageEvent, StorageType.SESSION);
      }
    
```